modules/users/tests/client/authentication.client.controller.tests.js

```diff
@@ -59,6 +60,28 @@
           expect($location.url()).toEqual('/');
         });
 
+        it('should be redirected to previous state after successful login',
+          inject(function (_$state_) {
+            $state = _$state_;
+            $state.previous = {
+              state: {
+                name: 'articles.create'
+              },
+              params: {},
+              href: '/articles/create'
+            };
+
+            // Test expected GET request
+            $httpBackend.when('POST', '/api/auth/signin').respond(200, 'Fred');
+
+            scope.signin(true);
+            $httpBackend.flush();
+            spyOn($state, 'transitionTo');
```

Comment:
This should be set before `scope.signin(true)` (line 77). However, even after that modification, the test doesn't pass. The actuals returned from the below `expect` don't match up with the previous state defined in this test. Upon further investigation, I logged `$state.previous` to the console in the signin method of the AuthenticationController & it was equal to the previous state defined here in this test.

Anyone else wanna take a shot at this? I'll keep working on it, but probably could use a fresh set of eyes.